### ISSUE TITLE
multi-argument modify

### ISSUE THREAD

**Issue by aplavin** [JuliaObjects/Accessors.jl#130]
I knew I saw it somewhere before but forgot, and now found a discussion by @jw3126 in the Setfield repo – https://github.com/jw3126/Setfield.jl/issues/137 :)

I added this feature to AccessorsExtra a few month ago, turns out to be useful sometimes.
Examples:
```julia
julia> old = (a=(1,), b=(2, 3))
julia> new = (a=(10,), b=(2, 0.3))
julia> modify(/, old, (@o _[∗][∗]), new)
(a = (0.1,), b = (1.0, 10.0))

julia> acc = (a=10, b=[(d=3, e=40)], c="x")
julia> newval = (a=1, b=[(d=30, e=4)], c="y")
julia> acc = modify(max, acc, RecursiveOfType(Number), newval)
(a = 10, b = [(d = 30, e = 40)], c="x")
```
Implementation is very simple in https://gitlab.com/aplavin/AccessorsExtra.jl/-/blob/master/src/modifymany.jl. Most of the lines are to handle Elements and Properties.

First, I wonder if you have any suggestions on the interface. Is that useful for what you had in mind?
Second, if this is considered generally within the Accessors scope, we can add it here. Wonder if this is widely useful (I think so, but not without promoting these features...), and worry a bit about "feature creep" of such a foundational package. In comparison, AccessorsExtra contains less clear-cut stuff anyways, so I'm much less worried there :)

**Comment by jw3126** [comment]
Yes, I think we can have this. I don't love the options we have for the signature though:
```julia
modify(f, optic, objs...) # clashes with existing modify(f, optic, obj)
modify(f, obj1, optic, objs...) # a bit ugly to have the optic in between
modify(f, objs..., optic) # a bit cumbersome for dispatch
```
Alternative we could use a new function for this.

**Comment by jw3126** [comment]
That is a good point, now I am more happy with the signature `modify(f, obj, optic, objs...)`.
